atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1851,13 +1851,12 @@ private void internalPutMetadataForTables(
         final Map<Cell, byte[]> newMetadata = Maps.newHashMap();
         final Collection<CfDef> updatedCfs = Lists.newArrayList();
         for (Entry<Cell, byte[]> entry : metadataRequestedForUpdate.entrySet()) {
-            Value val = persistedMetadata.get(entry.getKey());
-            if (val == null || !Arrays.equals(val.getContents(), entry.getValue())) {
+            if (newMetadataFound(persistedMetadata.get(entry.getKey()), entry.getValue())) {
                 newMetadata.put(entry.getKey(), entry.getValue());
-                updatedCfs.add(getCfForTable(
-                        tableReferenceFromBytes(entry.getKey().getRowName()),
-                        entry.getValue()));
             }
+            updatedCfs.add(getCfForTable(
+                    tableReferenceFromBytes(entry.getKey().getRowName()),
+                    entry.getValue()));
```

Comment:
I agree. @hsaraogi can we discuss why this does or doesn't solve the problem? I think what we want is for the `updatedCfs` to stay in the if statement but for the check to change. Instead of `newMetadataFound` we likely want `newOrDifferentAndChangeableMetadataFound`, so if it's brand new add it to updated and if it's a specific type of metadata change that we're cool with updating live then also update it.